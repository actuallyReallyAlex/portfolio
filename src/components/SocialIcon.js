import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'grommet'
import * as icons from 'grommet-icons'
import { useHover } from '../hooks/useHover'

const SocialIcon = ({ icon, link }) => {
  const Icon = icons[icon]
  const [hoverRef, isHovered] = useHover()

  return (
    <Button
      icon={
        <Icon
          className="slow-transition"
          color={
            isHovered ? 'rgba(255, 255, 255, 1)' : 'rgba(255, 255, 255, 0.75)'
          }
        />
      }
      onClick={() => window.open(link, '_blank')}
      ref={hoverRef}
    />
  )
}

SocialIcon.propTypes = {
  icon: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired
}

export default SocialIcon
