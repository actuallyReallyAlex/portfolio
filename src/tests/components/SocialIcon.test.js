import React from 'react'
import SocialIcon from '../../components/SocialIcon'
import { shallow } from 'enzyme'

const defaultProps = {
  icon: 'Code',
  link: 'www.google.com',
  size: 'medium'
}

describe('<SocialIcon />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <SocialIcon /> component.', () => {
    const component = shallow(<SocialIcon {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
