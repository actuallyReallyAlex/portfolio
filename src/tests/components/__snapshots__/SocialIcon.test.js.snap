// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialIcon /> Should render the <SocialIcon /> component. 1`] = `ShallowWrapper {}`;
